File: app/src/main/java/tetramax/android/MainActivity.java
package tetramax.android;

import android.content.res.AssetFileDescriptor;
import android.media.MediaPlayer;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = MainActivity.class.getSimpleName();

    private TextView musicInfoTextView;
    private Button playButton, stopButton, startServiceButton, stopServiceButton;

    private final MediaPlayer player = new MediaPlayer();
    private final Random random = new Random();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.i(TAG, "onCreate()");
        setContentView(R.layout.activity_main);

        playButton = findViewById(R.id.playButton);
        stopButton = findViewById(R.id.stopButton);
        musicInfoTextView = findViewById(R.id.musicInfoTextView);
        startServiceButton = findViewById(R.id.startServiceButton);
        stopServiceButton = findViewById(R.id.stopServiceButton);

        playButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                play();
            }
        });
        stopButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                stop();
            }
        });
        startServiceButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(
                        getApplicationContext(),
                        "Start service button will be used in the service implementation.",
                        Toast.LENGTH_SHORT).show();
            }
        });
        stopServiceButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(
                        getApplicationContext(),
                        "Stop service button will be used in the service implementation.",
                        Toast.LENGTH_SHORT).show();
            }
        });
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.i(TAG, "onStart()");
    }

    @Override
    protected void onStop() {
        Log.i(TAG, "onStop()");
        player.release();
        super.onStop();
    }

    public void play() {
        if (player.isPlaying()) {
            return;
        }
        final List<String> files = getFiles();
        final String song = files.get(random.nextInt(files.size()));

        Log.i(TAG, "Playing song " + song);

        try {
            final AssetFileDescriptor descriptor = getAssets().openFd(song);
            player.setDataSource(
                    descriptor.getFileDescriptor(),
                    descriptor.getStartOffset(),
                    descriptor.getLength()
            );
            descriptor.close();
            player.prepare();
        } catch (IOException e) {
            Log.w(TAG, "Could not open file", e);
            return;
        }

        player.setLooping(true);
        player.start();

        // display song info
        musicInfoTextView.setText(song);
    }

    public void stop() {
        if (player.isPlaying()) {
            player.stop();
        }

        player.reset();

        // display song info
        musicInfoTextView.setText("");
    }

    /**
     * Returns the list of mp3 files in the assets folder
     *
     * @return
     */
    private List<String> getFiles() {
        final List<String> mp3s = new ArrayList<>();
        try {
            final String[] files = getAssets().list("");
            if (files == null) {
                return mp3s;
            }

            for (String fileName : files) {
                if (fileName.toLowerCase().endsWith("mp3")) {
                    mp3s.add(fileName);
                }
            }
        } catch (IOException e) {
            Log.w(TAG, "Error while getting files.", e);
        }

        return mp3s;
    }
}

File: app/src/main/java/tetramax/android/MainActivity.kt
package tetramax.android

import android.content.Intent
import android.media.MediaPlayer
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import java.io.IOException
import java.util.*

class MainActivity : AppCompatActivity() {
    private var musicInfoTextView: TextView? = null
    private var startServiceButton: Button? = null
    private var stopServiceButton: Button? = null
    private var aboutButton: Button? = null

    private var player: MediaPlayer? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.i(TAG, "onCreate()")
        setContentView(R.layout.activity_main)
        musicInfoTextView = findViewById(R.id.musicInfoTextView)
        startServiceButton = findViewById(R.id.startServiceButton)
        stopServiceButton = findViewById(R.id.stopServiceButton)
        aboutButton = findViewById(R.id.aboutButton)

        startServiceButton?.setOnClickListener {
            Toast.makeText(
                applicationContext,
                "Start service button will be used in the service implementation.",
                Toast.LENGTH_SHORT
            ).show()
        }
        stopServiceButton?.setOnClickListener {
            Toast.makeText(
                applicationContext,
                "Stop service button will be used in the service implementation.",
                Toast.LENGTH_SHORT
            ).show()
        }
        aboutButton?.setOnClickListener {
            startActivity(
                Intent(
                    this@MainActivity,
                    AboutActivity::class.java
                )
            )
        }
    }

    override fun onStart() {
        super.onStart()
        player = MediaPlayer()
        Log.i(TAG, "onStart()")
    }

    /**
     * Starts the music player playback
     */
    fun play() {
        player?.let {
            if (it.isPlaying) {
                return
            }
            getFiles().random().apply {
                try {
                    val descriptor = assets.openFd(this)
                    it.setDataSource(
                        descriptor.fileDescriptor,
                        descriptor.startOffset,
                        descriptor.length
                    )
                    descriptor.close()
                    it.prepare()
                } catch (e: IOException) {
                    Log.w(TAG, "Could not open file", e)
                    return
                }
                it.isLooping = true
                it.start()

                // display song info
                musicInfoTextView?.text = this
                Log.i(TAG, "Playing song $this")
            }
        }
    }

    /**
     * Stops the music player playback
     */
    fun stop() {
        player?.let {
            if (it.isPlaying) {
                it.stop()
            }
            it.reset()
        }

        // display song info
        musicInfoTextView?.text = ""
    }

    /**
     * Returns the list of mp3 files in the assets folder
     *
     * @return
     */
    private fun getFiles(): List<String> =
        assets.list("")?.filter { it.toLowerCase(Locale.getDefault()).endsWith("mp3") }
            ?: emptyList()

    companion object {
        private val TAG = MainActivity::class.java.simpleName
    }
}